### PR TITLE
Pass camera and microphone lists to the exposure decision algorithm

### PR DESCRIPTION
**Pull request by youennf**
Loop through discovered devices first to build a microphone list.
Loop through discovered devices second to build a camera list.
Append microphone list then camera list to resultList.
Then use exposure decision algorithm for other devices with microphone list and camera list as input.

### CODE REVIEW

**Comment by alvestrand** [comment]
This PR imposes a requirement that all cameras come first, then all microphones.
It will make implementations that return cameras and microphones interleaved non-conformant, and will move the default microphone from "one of the two first entries" to "the position after all the cameras".
Why?

(I am very sensitive to changing specs to require things that are not currently required where existing implementations have been written without knowledge of these new requirements.)

**Comment by youennf** [comment]
> This PR imposes a requirement that all cameras come first, then all microphones.

This is the reverse actually that this PR intends: microphone first, then camera, then other devices.
This matches what Chrome and Safari do. Firefox also sorts device per kind but put camera first.

> It will make implementations that return cameras and microphones interleaved non-conformant,

Correct, though browsers I tested are not doing that.
There is a compat risk for user agents to interleave devices.

> will move the default microphone from "one of the two first entries" to "the position after all the cameras".

PR intent is to put default microphone as the first entry (but default camera further away).
No browser I tested implements the current wording of the spec.

> Why?

In general, I think specifying order will make developer's life easier.

We can probably mandate per-device-kind sorting as this seems already supported by implementations.
We need to do something about the default device since no browser implements the spec.

I also hope implementations can agree to a particular kind order (microphone first, camera second).
Firefox point of view is especially useful here since this PR requires a change on their end.

**Comment by youennf** [comment]
@alvestrand, for context, I updated https://github.com/w3c/mediacapture-output/pull/109 to use microphoneList based on this PR.

**Comment by henbos** [comment]
Looks good to me

**Comment by youennf** [comment]
@jan-ivar, can you look at the PR and see whether Firefox is fine aligning with it.

File: getusermedia.html [hunk 1]
```diff
@@ -2885,41 +2885,92 @@ <h2>Methods</h2>
...
+                        <li>
+                          <p>Add to <var>resultList</var> the first item of <var>microphoneList</var> if <var>microphoneList</var> is not empty.</p>
+                        </li>
+                        <li>
+                          <p>Add to <var>resultList</var> the first item of <var>cameraList</var> if <var>cameraList</var> is not empty.</p>
+                        </li>
+                      </ol>
+                    </li>
+                    <li>
+                      <p>Else, run the following steps:<p>
```
Comment (jan-ivar) [inline getusermedia.html:2945]:
```suggestion
                      <p>Otherwise, run the following steps:<p>
```

**Pull request merged**